cmake: Clean-up pass and code optimization to prepare for Linux update

PatTheMav opened the pull request:
### Description
Cleans up the CMake code-base based on experience after the recent CMake build system updates:

* Change OS-specific code branches to platform-specific generator expressions
* Enforce list sorting in some places to ensure consistency
* Disable CMake formatting for custom functions that are not recognised by `cmake-format` or for updated CMake functions whose updated signatures it doesn't support
* Switch interface libraries to object libraries where possible
* Remove custom CMake policy code for policies that were set to `NEW` by CMake 3.22+ anyway.
* Remove support for Qt5 in dependency resolution
* FFmpeg package selection uses an explicit version parameter (6.0.0+ on Windows and macOS, 5.1.0+ on Ubuntu) 
* Windows 🪟: Update Windows warning suppressions by investigating their root causes (some warnings' root causes need to be fixed upstream) and making suppressions more specific
* Windows 🪟: Enable fallback configurations for libraries' optimised build types
* macOS 🍏: Remove support for any generator besides Xcode on macOS

> [!IMPORTANT]
> 
> An accompanying update of `obs-browser` will also 
requires the CEF wrapper library to be re-built for Windows following the proper procedure outlined in the documentation: https://bitbucket.org/chromiumembedded/cef/wiki/LinkingDifferentRunTimeLibraries.md
>
> Without updating the submodule the project will still build without issues.

### Motivation and Context
Moving platform-specific source file and library dependency information into global blocks with generator expressions makes the requirements more explicit (and also groups all source code dependencies into one logical place), removing the need to search for their inclusion in other parts of the file (and reduces amount if code branches, which also improves `cmake-lint` checks).

Changes are mostly cosmetic or attempt to reduce complexity by removing seldom-used options or features introduced in recent updates. Windows-specific changes also aim to make more informed decisions about which warnings are suppressed and why.

### How Has This Been Tested?
Preliminary tests looked fine, specific tests are still pending:

- [x] Builds on Windows 11 (Browser disabled)
- [x] Builds on Windows 11 (With fixed runtime library for CEF-wrapper)
- [x] Builds on Windows 11 (CMake Legacy, Browser disabled)
- [x] Builds on Windows 11 (CMake Legacy, with fixed runtime library for CEF-wrapper)
- [x] Builds on macOS 13
- [x] Builds on macOS 14
- [x] Builds on Ubuntu 22.04 (CMake Legacy, AJA+WebRTC+New MPEGTS output disabled)
- [x] Builds on Ubuntu 23.04 (CMake Legacy)
- [x] Builds on FreeBSD 13.2 (CMake Legacy, AJA+JACK disabled)

### Types of changes
- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

tytan652 commented:
Some changes might around UI files conflicts with #9350.

RytoEX commented on plugins/text-freetype2/CMakeLists.txt:
```diff
@@ -11,27 +11,49 @@ endif()
+          $<$<PLATFORM_ID:Linux,FreeBSD>:Fontconfig::Fontconfig>)
+# cmake-format: on
+
+# freetype is hardcoded with /DEFAULTLIB:MSVCRT
```
Is this our fault when building? Or is this in the dependency itself?

PatTheMav replied:
I think I haven't found an official configuration option to change/adjust this in `freetype`.

RytoEX replied:
Well, the comment here will have to suffice as a sort of TODO for us for later.

RytoEX commented on plugins/win-capture/get-graphics-offsets/CMakeLists.txt:
```diff
@@ -4,8 +4,6 @@ add_library(_get-graphics-offsets INTERFACE)
 target_sources(_get-graphics-offsets INTERFACE d3d8-offsets.cpp d3d9-offsets.cpp dxgi-offsets.cpp
                                                get-graphics-offsets.c get-graphics-offsets.h)
 
-target_compile_options(_get-graphics-offsets INTERFACE $<IF:$<CONFIG:Debug>,/MTd,/MT>)
```
No longer needed?

PatTheMav replied:
At least I've encountered no compile errors in Debug or Release without it in Visual Studio 17.

notr1ch replied:
This is for ensuring that the standalone program does not depend on the dynamic CRT. I'm not entirely sure why we do that with our standalone helper programs given OBS itself requires the same CRT.

PatTheMav replied:
Got it, in that case the "cleaner" approach is to set the `MSVC_RUNTIME_LIBRARY` target property to `"MultiThreaded$<$<CONFIG:Debug>:Debug>"` (which is the officially suggested way since CMake 3.5).

RytoEX commented on plugins/win-capture/graphics-hook/CMakeLists.txt:
```diff
@@ -15,12 +15,13 @@ add_library(_graphics-hook INTERFACE)
-  INTERFACE d3d8-capture.cpp
-            d3d9-capture.cpp
-            d3d9-patches.hpp
+  INTERFACE # cmake-format: sortable
             d3d10-capture.cpp
             d3d11-capture.cpp
             d3d12-capture.cpp
+            d3d8-capture.cpp
+            d3d9-capture.cpp
+            d3d9-patches.hpp
```
This sorting is unfortunate.

PatTheMav replied:
Indeed, but it is correct given a character-based sorting algorithm.

RytoEX replied:
I suppose we either sort it manually, or just take this change as-is.

PatTheMav replied:
We kinda accepted the "unnatural" sorting introduced by `cmake-format` in other places, so accepting it here would just be consistent. 😄

RytoEX replied:
It's fine.

RytoEX commented on plugins/win-capture/graphics-hook/CMakeLists.txt:
```diff
@@ -29,10 +30,10 @@ target_sources(
             graphics-hook.rc)
 
 target_compile_definitions(_graphics-hook INTERFACE COMPILE_D3D12_HOOK)
-target_compile_options(_graphics-hook INTERFACE $<IF:$<CONFIG:Debug>,/MTd,/MT>)
```
No longer needed?

PatTheMav replied:
See above, I don't encounter any compile errors with this removed. This switch was usually added because of runtime library mismatches.

It could have been implicitly fixed by `ipc-util` having become an interface library and thus its sources are compiled with the actual target using the same runtime.

RytoEX commented on plugins/win-capture/inject-helper/CMakeLists.txt:
```diff
@@ -5,8 +5,9 @@ if(NOT TARGET OBS::obfuscate)
 add_library(_inject-helper INTERFACE)
+
 target_sources(_inject-helper INTERFACE inject-helper.c)
-target_compile_options(_inject-helper INTERFACE $<IF:$<CONFIG:Debug>,/MTd,/MT>)
```
No longer needed?

PatTheMav replied:
As per the comment above, I'll change this to the target property `MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>"`.

RytoEX commented on plugins/win-dshow/virtualcam-module/CMakeLists.txt:
```diff
@@ -57,11 +57,12 @@ if(OBS_CMAKE_VERSION VERSION_GREATER_EQUAL 3.0.0)
+              placeholder.cpp sleepto.c sleepto.h virtualcam-filter.cpp virtualcam-filter.hpp virtualcam-module.cpp)
   target_include_directories(_virtualcam INTERFACE "${CMAKE_CURRENT_BINARY_DIR}")
   target_compile_definitions(_virtualcam INTERFACE VIRTUALCAM_AVAILABLE)
-  target_compile_options(_virtualcam INTERFACE $<IF:$<CONFIG:Debug>,/MTd,/MT>)
```
No longer needed?

PatTheMav replied:
As per the comment above, I'll change this to the target property `MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>"`.

RytoEX reviewed (changes requested):
Mostly questions seeking rationale and clarification. Builds locally on Windows 11. Haven't done any further in-depth testing.

PatTheMav commented on UI/cmake/ui-qt.cmake:
```diff
@@ -1,4 +1,10 @@
+find_package(Qt6 REQUIRED Widgets Network Svg Xml)
+# cmake-format: on
+
+if(NOT OS_WINDOWS AND NOT OS_MACOS)
```
Because I'm not aware of AJA compatibility/availability for FreeBSD.

RytoEX replied:
Got it.

PatTheMav commented on UI/frontend-plugins/aja-output-ui/CMakeLists.txt:
```diff
@@ -8,80 +8,65 @@ if(NOT ENABLE_AJA)
+          $<$<PLATFORM_ID:Windows>:Winmm.lib>
+          $<$<PLATFORM_ID:Windows>:netapi32.lib>
+          $<$<PLATFORM_ID:Windows>:Shlwapi.lib>
+          "$<$<PLATFORM_ID:Darwin>:$<LINK_LIBRARY:FRAMEWORK,AppKit.framework>>"
+          "$<$<PLATFORM_ID:Darwin>:$<LINK_LIBRARY:FRAMEWORK,CoreFoundation.framework>>"
+          "$<$<PLATFORM_ID:Darwin>:$<LINK_LIBRARY:FRAMEWORK,IOKit.framework>>"
```
Because that generator expression resolves into the absolute path of the framework file and not a target name, so the "paths in quotes" rule applies.

RytoEX replied:
Got it.

PatTheMav commented on plugins/aja/CMakeLists.txt:
```diff
@@ -14,53 +14,68 @@ find_package(LibAJANTV2 REQUIRED)
   configure_file(cmake/windows/obs-module.rc.in win-aja.rc)
   target_sources(aja PRIVATE win-aja.rc)
-
-  target_compile_options(aja PRIVATE /wd4996)
```
Correct, the `COMPILE_WARNING_AS_ERROR` property is set on the target to allow the deprecation warnings to be emitted as actual warnings.

RytoEX replied:
Ah, I see now. Hopefully those warnings aren't too noisy.

PatTheMav commented on UI/cmake/os-windows.cmake:
```diff
@@ -28,19 +30,21 @@ target_sources(
                                          nlohmann_json::nlohmann_json Detours::Detours)
+
 target_compile_definitions(obs-studio PRIVATE PSAPI_VERSION=2)
-target_link_options(obs-studio PRIVATE /IGNORE:4098 /IGNORE:4099)
```
Correct, instead of suppressing it globally I investigated the root causes and  selectively disabled the warning because it is only causes by a small subset of dependencies:

* `jansson` needs to be built with `JANSSON_STATIC_CRT`
  * Affected modules are `win-capture`, `rtmp-services`, `obs-outputs`, `ftl-sdk`, and `libobs`
* `zstd` seems to be hardcoded with `/DEFAULTLIB:LIBCMT`
  * Only affected module is `win-updater`
* `freetype` is hardcoded with `/DEFAULTLIB:MSVCRT`
  * Only affected module is `text-freetype2` 

So far it looks like only `jansson` can be compiled differently to fix the root cause, but once that is done we'd be left with `win-updater` and `text-freetype2` stuck with dependencies that use the "wrong" runtime libraries.

RytoEX replied:
Got it.

On the topic of Jansson, IIRC, setting `JANSSON_STATIC_CRT` ON causes it to use `/MT` and `/MTd`, which we found undesirable when we added Jansson to obs-deps. See https://github.com/obsproject/obs-deps/commit/0964f5e8eef57125b86d73799e2b5c7ea17d169a.

PatTheMav commented on UI/cmake/feature-youtube.cmake:
```diff
@@ -1,7 +1,8 @@
    AND YOUTUBE_CLIENTID_HASH MATCHES "^(0|[a-fA-F0-9]+)$"
-   AND YOUTUBE_SECRET_HASH MATCHES "^(0|[a-fA-F0-9]+)$")
+   AND YOUTUBE_SECRET_HASH MATCHES "^(0|[a-fA-F0-9]+)$"
+   AND TARGET OBS::browser-panels)
```
Twitch and YouTube service integrations are directly tied to associated browser docks, so it's a necessary dependency.

RytoEX replied:
The YouTube integration doesn't require browser panels for just the account connection, IIRC. Its secondary features, such as Chat, do, but the connection itself does not. Twitch's integration does require browser panels for all parts.

PatTheMav replied:
That might have been true until the recent update by YouTube, which requires browser docks to be present regardless of whether a service connection exists (the dock is added and enabled by default just by selecting YouTube as a service).

RytoEX replied:
IIRC, the YouTube Live Control Room dock does not require being connected via the YouTube integration, so it is technically separate.

PatTheMav replied:
I think _not_ adding the requirement lead to some breakage if the browser source is not available, but cannot replicate the issue anymore.

tytan652 replied:
Normally, adding YouTube API key and secret with Browser disables leads to build issue because of how the new dock was added to CMake (and even OBS).

PatTheMav replied:
> Normally, adding YouTube API key and secret with obs-browser disabled leads to build issues because of how the new dock was added to CMake (and even OBS).
> 
> Edit: Did not see the last edit.

Yeah I couldn't remember at first, but then went in and tried to compile in a slow VM, took some time to get the error. 😄

RytoEX replied:
Ah, I think I see what you mean.

PatTheMav commented on UI/frontend-plugins/decklink-output-ui/CMakeLists.txt:
```diff
@@ -7,54 +7,44 @@ if(NOT ENABLE_DECKLINK)
+  PRIVATE OBS::libobs
+          OBS::frontend-api
+          OBS::ui-support
+          Qt::Widgets
```
I have no real opinion on that, would be fine with either.

RytoEX replied:
For now, we can leave the changes as-is and make that change separately later, if we find it desirable.

PatTheMav commented on UI/obs-frontend-api/CMakeLists.txt:
```diff
@@ -18,6 +18,8 @@ elseif(OS_MACOS)
 
-set_target_properties_obs(obs-frontend-api PROPERTIES FOLDER frontend PREFIX "" PUBLIC_HEADER obs-frontend-api.h)
-
 target_export(obs-frontend-api)
+
+# cmake-format: off
+set_target_properties_obs(obs-frontend-api PROPERTIES FOLDER frontend PREFIX "" PUBLIC_HEADER obs-frontend-api.h)
+# cmake-format: on
```
Correct.

RytoEX replied:
Got it.

PatTheMav commented on UI/win-update/updater/CMakeLists.txt:
```diff
@@ -22,13 +22,14 @@ target_sources(
           updater.manifest
           updater.rc)
 
-target_compile_options(updater PRIVATE $<IF:$<CONFIG:DEBUG>,/MTd,/MT> "/utf-8")
```
The `DEBUG` config portion was just required to fix `zstd`s hardcoded runtime library mismatch, for which a linker option is used instead.

RytoEX replied:
Do you mean the `/IGNORE:4098` below? And did I understand the `"/utf-8"` part correctly?

PatTheMav replied:
Correct, `/utf-8` is enabled globally, the runtime library switch is unnecessary because of the silencing of `4098`.

But I'm starting to gravitate towards replacing that with `/NODEFAULTLIB:<library>` instead to be specific about which "wrong" default library definition to ignore.

RytoEX replied:
I briefly looked into `/NODEFAULTLIB:<library>` and saw some cautions, but didn't try implementing it. We'll see how this goes.

PatTheMav commented on cmake/macos/compilerconfig.cmake:
```diff
@@ -15,77 +22,57 @@ if(NOT CMAKE_OSX_ARCHITECTURES)
-  if(CMAKE_BUILD_TYPE IN_LIST _release_configs)
-    add_link_options(LINKER:-dead_strip)
-  endif()
+# Default ObjC++ compiler options used by Xcode: *  -Wno-non-virtual-dtor
```
Yeah, might require an empty comment line to prohibit it from doing that.

RytoEX replied:
Probably. Let's do that then.

PatTheMav commented on cmake/windows/helpers.cmake:
```diff
@@ -1,7 +1,8 @@
 
 # cmake-format: off
-# cmake-lint: disable=C0301
+# cmake-lint: disable=C0103
```
Correct, C0301 (line too long) was fixed, C103 usually complains about variable naming (e.g. using lowercase or underline-prefixed variables where CMake-Lint would prefer all uppercase).

RytoEX replied:
Got it.

PatTheMav commented on deps/CMakeLists.txt:
```diff
@@ -1,16 +1,14 @@
 if(OS_WINDOWS)
-  if(NOT MINGW)
-    add_subdirectory(w32-pthreads)
-  endif()
```
Correct, there is no MINGW-specific compiler configuration available and compatibility with anything but MSVC for Windows was absent in the codebase already.

RytoEX replied:
Got it.

PatTheMav commented on deps/blake2/CMakeLists.txt:
```diff
@@ -1,18 +1,13 @@
-                               "$<$<C_COMPILER_ID:AppleClang,Clang>:-Wno-error=implicit-int-conversion>")
-
-if(OS_WINDOWS)
-  target_compile_options(blake2 PRIVATE $<IF:$<CONFIG:Debug>,/MTd,/MT> /Zl)
-  target_compile_definitions(blake2 PRIVATE inline=_inline restrict=__restrict)
```
I've removed them optimistically and have met no compile or link errors when building the project.

From the little information I was able to dig up, this might have been necessary for quite ancient MSVC versions.

RytoEX replied:
I took a quick look and that seems to be the case. It seemed to define `inline` as `_inline` and `restrict` as `__restrict`? MSVC should support `inline` just fine these days, and blake2-impl.h does its own detection for `inline` support, which shows it setting `BLAKE2_INLINE` to `inline` in VS2022 17.7.

I don't see any uses of `restrict` in the BLAKE2 files.

@notr1ch ?

https://github.com/obsproject/obs-studio/commit/23a115b0b3f8215cc63799ab6256c4f28872230c#diff-809b55eb4b1630c733648be54d6d7c9b934eb0671e97ede6b82506ca0481242fR18-R20

RytoEX replied:
This is probably fine.

PatTheMav commented on deps/glad/CMakeLists.txt:
```diff
@@ -2,29 +2,32 @@ cmake_minimum_required(VERSION 3.22...3.25)
+          $<$<PLATFORM_ID:Windows>:src/glad_wgl.c>
+          $<$<TARGET_EXISTS:OpenGL::EGL>:src/glad_egl.c>
+          $<$<TARGET_EXISTS:OpenGL::EGL>:include/EGL/eglplatform.h>
+  PUBLIC include/glad/glad.h
+         "$<$<PLATFORM_ID:Windows>:${CMAKE_CURRENT_SOURCE_DIR}/include/glad/glad_wgl.h>"
+         "$<$<TARGET_EXISTS:OpenGL::EGL>:${CMAKE_CURRENT_SOURCE_DIR}/include/glad/glad_egl.h>")
```
Usually header files don't need to be provided to the target's sources property at all (the only reason to do so is for visibility in Visual Studio/CMake).

Specifying the `PUBLIC` include directory is sufficient to allow the preprocessor to resolve `#include <glad/glad.h>`, but marking the headers makes it explicit that they are API headers.

RytoEX replied:
Got it.

PatTheMav commented on deps/glad/CMakeLists.txt:
```diff
@@ -2,29 +2,32 @@ cmake_minimum_required(VERSION 3.22...3.25)
+target_link_libraries(
+  obsglad
+  PRIVATE $<$<NOT:$<PLATFORM_ID:Windows,Darwin>>:${CMAKE_DL_LIBS}>
+  PUBLIC OpenGL::GL $<$<TARGET_EXISTS:OpenGL::EGL>:OpenGL::EGL>)
```
This is necessary because `glad` is not a static library anymore, but an `OBJECT` library - it's compiled object files are added directly to the linker command line for `libobs-opengl` and thus its dependencies need to be transitive as well.

RytoEX replied:
Got it.

PatTheMav commented on deps/ipc-util/CMakeLists.txt:
```diff
@@ -1,14 +1,12 @@
-target_sources(ipc-util PRIVATE ipc-util/pipe.h)
-target_include_directories(ipc-util PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
+target_sources(
+  ipc-util INTERFACE "$<$<PLATFORM_ID:Windows>:${CMAKE_CURRENT_SOURCE_DIR}/ipc-util/pipe-windows.c>" ipc-util/pipe.h
+                     "$<$<PLATFORM_ID:Windows>:${CMAKE_CURRENT_SOURCE_DIR}/ipc-util/pipe-windows.h>")
```
The existence of `pipe-posix.c` and `pipe-posix.h` suggested to me that there was at least the idea of `ipc-util` being cross platform.

RytoEX replied:
I'm not sure. Would probably have to do some digging, but this change is probably fine.

PatTheMav replied:
It is functionally the same, but it is effectively only used by `win-capture` and `graphics-hook`.

Alternatively we could remove the generator expressions and remove `pipe-posix.h` and `pipe-posix.c` as well as the non-WIN32 specific preprocessor definition and make it a Windows-specific dependency.

PatTheMav commented on deps/json11/CMakeLists.txt:
```diff
@@ -1,9 +1,17 @@
-target_sources(json11 INTERFACE json11.cpp json11.hpp)
+target_include_directories(json11 PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "(AppleClang|Clang)" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 14)
+  target_compile_options(json11 PUBLIC -Wno-unqualified-std-cast-call)
+endif()
```
Correct, it would've been quite a long expression and became too unwieldy to parse for my tastes.

RytoEX replied:
Understood.

PatTheMav commented on deps/libcaption/CMakeLists.txt:
```diff
@@ -1,37 +1,41 @@
           src/scc.c
+          src/srt.c
           src/utf8.c
+          src/vtt.c
```
Correct, I've actually hit compile errors without this change now that it is an `OBJECT` library iirc.

RytoEX replied:
Huh. Weird. The file was added quite some time ago, but it wasn't added to the CMake:
https://github.com/obsproject/obs-studio/commit/a8517f3698e5e3d99335c97b9876e38a4612fac1

Looks like dvtcc.c is in the same situation. Don't know why they ended up that way or why you're getting compile errors now.

PatTheMav replied:
Not adding this file leads to linker errors:

```
[ 38%] Linking C executable obs-ffmpeg-mux
/usr/bin/ld: ../../../libobs/libobs.so.30: undefined reference to `vtt_new'
/usr/bin/ld: ../../../libobs/libobs.so.30: undefined reference to `_vtt_parse'
/usr/bin/ld: ../../../libobs/libobs.so.30: undefined reference to `vtt_free'
collect2: error: ld returned 1 exit status
```

From what I can tell this is due to the change to an `OBJECT` library - `libobs` pulls in the object files and thus also the header, which exposes these symbols. Changing it back into a static library seems to be the "cleaner" alternative.

PatTheMav commented on deps/media-playback/CMakeLists.txt:
```diff
@@ -1,24 +1,27 @@
 cmake_minimum_required(VERSION 3.22...3.25)
 
-find_package(FFmpeg REQUIRED COMPONENTS avcodec avdevice avutil avformat)
+find_package(FFmpeg REQUIRED avcodec avdevice avutil avformat)
```
I just prefer the shorter form.

RytoEX replied:
I'll mull it over. I'm not super opposed to the short form.

PatTheMav commented on deps/obs-scripting/obslua/CMakeLists.txt:
```diff
@@ -25,26 +25,28 @@ swig_add_library(
+          $<$<PLATFORM_ID:Windows>:/wd4267>
+          $<$<C_COMPILER_ID:AppleClang,Clang>:-Wno-shorten-64-to-32>
+          $<$<C_COMPILER_ID:AppleClang,Clang>:-Wno-unreachable-code>
+          $<$<C_COMPILER_ID:GNU>:-Wno-maybe-uninitialized>)
```
At least in my checks GCC was the only one who emitted that warning.

RytoEX replied:
To clarify, I asked this because the section it replaced was:
```
elseif(OS_LINUX OR OS_FREEBSD)
  target_compile_options(obslua PRIVATE -Wno-maybe-uninitialized)
```

So I was wondering if it's safe to assume that only GCC is used on those systems (or do people sometimes use clang on those?) _and_ that it's the only compiler (that we cover) that emits this warning. If GCC is the only compiler that emits the warning, then this is fine.

PatTheMav replied:
The `maybe-uninitialized` warning is not supported by `clang` (it's a GCC-only warning) and was present in the old code. It does not seem to be emitted by the code generated by `swig` anymore, but if we wish to retain it, it needs to be used for GCC only.

RytoEX replied:
Got it.

PatTheMav commented on deps/obs-scripting/obslua/CMakeLists.txt:
```diff
@@ -25,26 +25,28 @@ swig_add_library(
-target_compile_options(obslua PRIVATE "$<$<C_COMPILER_ID:AppleClang,Clang>:-Wno-error=shorten-64-to-32>")
+target_compile_options(
+  obslua
+  PRIVATE $<$<PLATFORM_ID:Windows>:/wd4054>
+          $<$<PLATFORM_ID:Windows>:/wd4197>
+          $<$<PLATFORM_ID:Windows>:/wd4244>
+          $<$<PLATFORM_ID:Windows>:/wd4267>
```
Yeah my intention was to have each warning be it's own "unit" or line. I also prefer actual list items vs semicolon-separated strings when providing an actual list of items.

Otherwise the generator expression would need to be wrapped in double quotes with a semicolon between each value, which worsens readability in my opinion.

RytoEX replied:
Got it. I find this a bit clearer too, so this can stay as-is.

PatTheMav commented on deps/obs-scripting/obspython/CMakeLists.txt:
```diff
@@ -35,29 +35,26 @@ file(
+  obspython PRIVATE "$<$<PLATFORM_ID:Windows>:${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG>>"
+                    "$<$<PLATFORM_ID:Darwin>:$<TARGET_PROPERTY:Python::Python,INTERFACE_INCLUDE_DIRECTORIES>>")
 
-list(APPEND _SWIG_DEFINITIONS "SWIG_TYPE_TABLE=obspython" "Py_ENABLE_SHARED=1" "SWIG_PYTHON_INTERPRETER_NO_DEBUG")
```
The original intention was to de-duplicate the definitions being provided to the compiler as well as the `swig` preprocessor, but the code already explicitly added the options twice.

Thus this line was superfluous.

RytoEX replied:
Got it.

PatTheMav commented on libobs-d3d11/CMakeLists.txt:
```diff
@@ -25,9 +25,9 @@ target_sources(
 
 configure_file(cmake/windows/obs-module.rc.in libobs-d3d11.rc)
 target_sources(libobs-d3d11 PRIVATE libobs-d3d11.rc)
-target_compile_options(libobs-d3d11 PRIVATE /wd4595)
```
The intention is to downgrade this warning back to an actual warning so that somebody can investigate the severity of the warning and fix the underlying error:

`warning C4595: 'operator new[]': non-member operator new or delete functions may not be declared inline`

Inlining `new` and `delete` is effectively "banned" in C++, so we should fix this instead:

> The program is ill-formed, no diagnostic required if more than one replacement is provided in the program for any of the replaceable allocation function, or if a replacement is declared with the inline specifier.

RytoEX replied:
Interesting. Wonder what happened there. Seems to originate from here unless I'm mistaken.
https://github.com/obsproject/obs-studio/commit/db895092ed60a96c64bdf06823ee144867c4e267#diff-31d6cb641dc2ed5bf5e13c32bd469d3b31464d8c80c9fde4f650fd367c849839R28

PatTheMav replied:
Correct, I originally just silenced it, but in the course of the cleanup I figured out that I can revert the "warnings as error" option as a target option (because MSVC does not have an equivalent to `-Wno-error=<warning>`).

RytoEX replied:
Got it.

PatTheMav commented on libobs-opengl/CMakeLists.txt:
```diff
@@ -26,51 +33,44 @@ target_sources(
+          OBS::glad
+          "$<$<PLATFORM_ID:Darwin>:$<LINK_LIBRARY:FRAMEWORK,Cocoa.framework>>"
+          "$<$<PLATFORM_ID:Darwin>:$<LINK_LIBRARY:FRAMEWORK,IOSurface.framework>>"
+          $<$<PLATFORM_ID:Linux,FreeBSD>:xcb::xcb>
+          $<$<PLATFORM_ID:Linux,FreeBSD>:X11::x11-xcb>
+          $<$<AND:$<PLATFORM_ID:Linux,FreeBSD>,$<BOOL:${ENABLE_WAYLAND}>>:OpenGL::EGL>
+          $<$<AND:$<PLATFORM_ID:Linux,FreeBSD>,$<BOOL:${ENABLE_WAYLAND}>>:Wayland::EGL>)
```
Indeed they will, because generator expressions are evaluated at the generation step _after_ the configuration step, which is one of their benefits.

RytoEX replied:
Interesting. The line order may be a little confusing to those who don't know that, but I suppose it's fine.

PatTheMav replied:
Logically it makes sense to put dependency checks at the top to "error out early" in the processing of the script though.

RytoEX replied:
Thanks. Appreciate the change.

PatTheMav commented on libobs/CMakeLists.txt:
```diff
@@ -2,28 +2,43 @@ cmake_minimum_required(VERSION 3.22...3.25)
+if(OS_WINDOWS OR OS_MACOS)
+  set(ffmpeg_version 6)
+else()
+  set(ffmpeg_version 5.1)
```
That would indeed be correct - this is adjusted in the PR that enables the non-legacy path for Linux, but could just be backported.

RytoEX replied:
I think it's better to just get it right in this PR, since we know that's the reality we currently support, instead of writing one value only to change it in the next PR.

PatTheMav commented on libobs/cmake/os-freebsd.cmake:
```diff
@@ -1,17 +1,18 @@
 find_package(x11-xcb REQUIRED)
 # cmake-format: off
-find_package(xcb OPTIONAL_COMPONENTS xcb-xinput QUIET)
+find_package(xcb REQUIRED xcb OPTIONAL_COMPONENTS xcb-xinput)
```
The non-legacy CMake codepath relies on CMake's own dependency resolution to report missing dependencies. The legacy code choses to suppress CMake's own errors and instead display custom error messages, unnecessarily adding code complexity.

In fact the legacy code would potentially configure the project without issue at first, but then throw errors at build generation because `XCB::XCB` was added as a dependency but the target definition could not be resolved.

RytoEX replied:
Got it.

PatTheMav commented on plugins/coreaudio-encoder/CMakeLists.txt:
```diff
@@ -13,20 +13,22 @@ endif()
 
 if(OS_WINDOWS)
   configure_file(cmake/windows/obs-module.rc.in coreaudio-encoder.rc)
   target_sources(coreaudio-encoder PRIVATE coreaudio-encoder.rc windows-imports.h)
```
Correct, looks like an oversight.

RytoEX replied:
Will wait for a follow-up.

RytoEX commented on UI/cmake/feature-macos-update.cmake:
```diff
@@ -2,10 +2,6 @@ include_guard(DIRECTORY)
 
 find_package(nlohmann_json REQUIRED)
 
-if(NOT TARGET OBS::blake2)
-  add_subdirectory("${CMAKE_SOURCE_DIR}/deps/blake2" "${CMAKE_BINARY_DIR}/deps/blake2")
-endif()
-
```
You'd probably have to ask @derrod for details.

derrod replied:
Nothing in `feature-macos-update.cmake` should require browser docks, but both WhatsNew and Sparkle rely on these components so if either is enabled this file needs to be included. The inclusion of blake2 was moved here because otherwise building OBS with Sparkle but without browser support would fail.

PatTheMav replied:
Ah right, because we need connected the Sparkle functionality to our branch management. Should just be changed to an include guard then to avoid the same file being included twice (as it does right now).

RytoEX replied:
Thanks for clarifying and resolving.

PatTheMav commented on plugins/mac-virtualcam/src/camera-extension/CMakeLists.txt:
```diff
@@ -13,7 +13,9 @@ foreach(_uuid IN ITEMS VIRTUALCAM_DEVICE_UUID VIRTUALCAM_SOURCE_UUID VIRTUALCAM_
 
-project(mac-camera-extension LANGUAGES Swift)
+enable_language(Swift)
+
+# project(mac-camera-extension LANGUAGES Swift)
```
Was replaced by `enable_language(Swift)` which _seems_ to work, so this could actually be removed.

RytoEX replied:
CMake 3.15 added support for "Swift" to `enable_language`.
> This is the same as the [project()](https://cmake.org/cmake/help/v3.27/command/project.html#command:project) command but does not create any of the extra variables that are created by the project command.

https://cmake.org/cmake/help/v3.27/command/enable_language.html

If it seems to work, then we can just remove the line(s) it replaced.

PatTheMav replied:
Yeah the only reason I originally added the `project` call was to badger CMake into recognising that we use Swift for the module - `enable_language` is the more appropriate function without the baggage of `project`.

PatTheMav commented on plugins/obs-filters/cmake/rnnoise.cmake:
```diff
@@ -1,46 +1,48 @@
+  else()
+    find_package(Librnnoise REQUIRED)
+
+    target_sources(obs-filters PRIVATE noise-suppress-filter.c)
```
It should, if I'm not mistaken. Is adjusted in the Linux-specific update because this code path is not taken by Linux yet anyway.

PatTheMav commented on plugins/obs-vst/CMakeLists.txt:
```diff
@@ -12,46 +12,48 @@ endif()
+  PRIVATE # cmake-format: sortable
+          $<$<PLATFORM_ID:Darwin>:mac/EditorWidget-osx.mm>
+          $<$<PLATFORM_ID:Darwin>:mac/VSTPlugin-osx.mm>
+          $<$<PLATFORM_ID:Linux,FreeBSD>:linux/EditorWidget-linux.cpp>
```
While this would only take effect after the Linux updates have been merged, it would actually be up to debate how "supported" OpenBSD actually is. Without an explicit list of supported features by the platform I'm currently not able to say which parts of OBS Studio are supported in OpenBSD at all.

RytoEX replied:
As far as I know, OBS on OpenBSD should operate much the same as on FreeBSD, with some exceptions.
See https://github.com/obsproject/obs-studio/commit/0d222b6b56fa3a49ed3a6604186870dd9f90e88b.

PatTheMav replied:
In that case I'll have to do another pass over the entire codebase here (and in the Linux update) to explicitly add OpenBSD or revert to a negative check ("not Windows and also not macOS") to lump in any other platform.

Otherwise we'd have to use `PLATFORM_ID`-based generator expressions for Windows and macOS, but a manual `BOOL`-based expression to check for any kind of Linux or Unix.

RytoEX replied:
Right. As mentioned, before we seem to have previously (currently?) put Linux, FreeBSD, and OpenBSD together in CMake as being `OS_POSIX` (`cmake/Modules/ObsHelpers.cmake`), but now the new CMake path (`cmake/common/osconfig.cmake`) creates:
* OS_LINUX
* OS_FREEBSD
* OS_OPENBSD

PatTheMav replied:
Updated to explicitly include all three of those.

RytoEX replied:
As an addendum, it looks like the old path actually set _both_ OS_POSIX _and_ OS_LINUX, OS_FREEBSD, and OS_OPENBSD. Is there a reason we omitted OS_POSIX from the new path?

PatTheMav replied:
`OS_POSIX` originally also included macOS and served as a general "not Windows" catch-all which was already changed when I made macOS a more distinct platform.

This change just continues that to make platform selection explicit across the board.

It also avoids conditionals like `if(OS_POSIX AND NOT OS_FREEBSD)` which seem less clear than `if(OS_LINUX AND OS_OPENBSD)` to me.

PatTheMav commented on plugins/obs-qsv11/CMakeLists.txt:
```diff
@@ -26,7 +34,19 @@ target_sources(
+          $<$<PLATFORM_ID:Linux>:Libva::va>
+          $<$<PLATFORM_ID:Linux>:Libva::drm>)
+
+target_link_options(obs-qsv11 PRIVATE $<$<PLATFORM_ID:Windows>:/IGNORE:4099>)
```
Yeah something messed up a merge somewhere.

PatTheMav commented on UI/frontend-plugins/aja-output-ui/CMakeLists.txt:
```diff
@@ -8,80 +8,65 @@ if(NOT ENABLE_AJA)
 if(OS_WINDOWS)
   configure_file(cmake/windows/obs-module.rc.in aja-output-ui.rc)
   target_sources(aja-output-ui PRIVATE aja-output-ui.rc)
-  target_compile_options(aja-output-ui PRIVATE /wd4996)
```
Correct, the warning is only emitted by the `aja` target (where warnings are downgraded from errors again).

RytoEX replied:
Got it.

PatTheMav commented on plugins/obs-outputs/cmake/ftl.cmake:
```diff
@@ -1,77 +1,67 @@
+          $<$<NOT:$<PLATFORM_ID:Windows>>:ftl-sdk/libftl/posix/socket.c>
+          $<$<NOT:$<PLATFORM_ID:Windows>>:ftl-sdk/libftl/posix/threads.c>
+          $<$<NOT:$<PLATFORM_ID:Windows>>:ftl-sdk/libftl/posix/threads.h>
+          $<$<PLATFORM_ID:Windows>:ftl-sdk/libftl/gettimeofday/gettimeofday.c>
```
Yeah I've literally just took the source file lists from the OS-specific conditional branches and added them via generator expression, so such oversights would have been inherited. I cleaned this up now.

RytoEX replied:
Works for me.

PatTheMav commented on plugins/obs-outputs/cmake/ftl.cmake:
```diff
@@ -1,77 +1,67 @@
-endif()
-
-set(target_headers ${target_sources})
-list(FILTER target_sources INCLUDE REGEX ".+ftl-sdk/.+\\.(m|c[cp]?p?|swift)")
-list(FILTER target_headers INCLUDE REGEX ".+ftl-sdk/.+\\.h(pp)?")
-
-source_group("ftl-sdk\\Source Files" FILES ${target_sources})
-source_group("ftl-sdk\\Header Files" FILES ${target_headers})
```
`ftl-sdk` is an `OBJECT` library now, so CMake generates a directory for it by itself.

`INTERFACE` libraries have no representation in the target tree (because its sources are added directly to the source list of the target that links the interface library), which is why this was needed before.

RytoEX replied:
Got it.

PatTheMav commented:
Force-pushed an update to address some reviews, also:

* Added explicit checks for `OS_OPENBSD` and `OpenBSD` platform id
* Switched from `C_COMPILE_ID` checks to `COMPILE_LANG_AND_ID`, which has the benefit of applying the expression when the _target's_ source code language and associated compiler id match (and not just if the compiler id for the language in general matches).
* Removed the addition if the `-py3` flag to `swig`, which has been deprecated a while ago.

PatTheMav commented on UI/cmake/ui-qt.cmake:
```diff
@@ -1,4 +1,10 @@
-find_qt(COMPONENTS Widgets Network Svg Xml COMPONENTS_LINUX Gui)
+# cmake-format: off
+find_package(Qt6 REQUIRED Widgets Network Svg Xml)
```
As with other places I just prefer using the shorthand, but have no hard objection to adding it. In any case I'd want this to be consistent across the codebase though.

PatTheMav replied:
LMK if you have come to a decision regarding the short-hand form, as I would need to update it throughout the code-base to ensure consistency then. Right now it should use the shortened form throughout.

RytoEX replied:
Let's just use the short form since that's already in this PR. Will keep the churn down and reduce chance of mistakes slipping in. We can always change it later if we find a reason to do so.

RytoEX commented on plugins/text-freetype2/CMakeLists.txt:
```diff
@@ -11,27 +11,52 @@ endif()
+          $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:Fontconfig::Fontconfig>)
+# cmake-format: on
+
+# Freetype is hard-coded with /DEFAULTLIB:MSVCRT
```
```suggestion
# FreeType is hard-coded with /DEFAULTLIB:MSVCRT
```
In descriptive text, the correct capitalization is FreeType.

PatTheMav commented on UI/cmake/feature-whatsnew.cmake:
```diff
@@ -1,9 +1,14 @@
 
 if(ENABLE_WHATSNEW AND TARGET OBS::browser-panels)
   if(OS_MACOS)
     include(cmake/feature-macos-update.cmake)
-  elseif(OS_LINUX)
+  elseif(
+    OS_LINUX
+    OR OS_FREEBSD
+    OR OS_OPENBSD)
```
Intentionally expanding based on the answer that BSDs should more or less be on the same level as Linux.

derrod replied:
Mostly just because this feature isn't super useful outside of official first-party builds, which we currently only provide for Ubuntu and Flatpak. I haven't tested it on other BSDs but I guess it should work?

PatTheMav replied:
You could argue it doesn't make sense on those platforms because while users might be informed about a new release of OBS Studio, its availability as a system package is dependent on the maintainer of a ports package (if there is any).

PatTheMav commented on UI/cmake/ui-qt.cmake:
```diff
@@ -1,4 +1,12 @@
+find_package(Qt6 REQUIRED Widgets Network Svg Xml)
+# cmake-format: on
+
+if(OS_LINUX
+   OR OS_FREEBSD
+   OR OS_OPENBSD)
+  find_package(Qt6 REQUIRED Gui)
+endif()
```
Just like with `OS_POSIX`' removal the intention is to make platform selection more explicit.

RytoEX replied:
I think for specific features, the more explicit selections are fine. For a dependency for the entire app, I'm not sure if we want it to be restrictive, or we may have to edit it later to be more inclusive. I'm open to considering adopting the more restrictive condition currently here, but I'll need to think about it.

PatTheMav replied:
My main thinking is that we currently support specific platforms and the code makes a varying amount of assumptions about that being the case.

The explicit selection has the benefit that project configuration will fail on unsupported platforms and will not surprisingly generate a non-working build project.

RytoEX replied:
I'll mull it over.

RytoEX replied:
I suppose this is fine. We'll worry about issues when they occur.

PatTheMav commented on UI/frontend-plugins/aja-output-ui/CMakeLists.txt:
```diff
@@ -8,80 +8,67 @@ if(NOT ENABLE_AJA)
+
+# cmake-format: off
+find_package(Qt6 REQUIRED Widgets)
+if(OS_LINUX)
+  find_package(Qt6 REQUIRED Gui)
+endif()
```
See above - though in this case this would require the feedback whether AJA is available on FreeBSD/OpenBSD.

PatTheMav commented on UI/frontend-plugins/aja-output-ui/CMakeLists.txt:
```diff
@@ -8,80 +8,67 @@ if(NOT ENABLE_AJA)
+endif()
+# cmake-format: on
+
+if(OS_LINUX
+   OR OS_FREEBSD
+   OR OS_OPENBSD)
+  find_package(X11 REQUIRED)
+endif()
```
In combination with the prior comment this is probably an oversight - both conditions should be aligned and should be based on whether AJA can actually be used on FreeBSD/OpenBSD.

PatTheMav commented on deps/libcaption/CMakeLists.txt:
```diff
@@ -5,33 +5,39 @@ add_library(OBS::caption ALIAS caption)
+# cmake-format: off
+target_compile_options(
+  caption
+  PRIVATE $<$<PLATFORM_ID:Windows>:/wd4267>
```
This might be superfluous now that `vtt.c` has been removed.

PatTheMav commented on libobs-opengl/CMakeLists.txt:
```diff
@@ -9,9 +9,31 @@ if(NOT TARGET OBS::glad)
   add_subdirectory("${CMAKE_SOURCE_DIR}/deps/glad" "${CMAKE_BINARY_DIR}/deps/glad")
 endif()
 
+if(OS_LINUX
+   OR OS_FREEBSD
+   OR OS_OPENBSD)
+  find_package(X11 REQUIRED)
+  # cmake-format: off
+  find_package(xcb REQUIRED xcb)
+  # cmake-format: on
+  find_package(x11-xcb REQUIRED)
```
Correct, wherever there was just FreeBSD included it was updated to also include OpenBSD.

RytoEX replied:
It does look like this previously used OS_POSIX, so this expansion is correct.

PatTheMav commented on libobs-opengl/CMakeLists.txt:
```diff
@@ -26,51 +48,33 @@ target_sources(
+          "$<$<PLATFORM_ID:Darwin>:$<LINK_LIBRARY:FRAMEWORK,IOSurface.framework>>"
+          $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:xcb::xcb>
+          $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:X11::x11-xcb>
+          $<$<AND:$<PLATFORM_ID:Linux,FreeBSD>,$<BOOL:${ENABLE_WAYLAND}>>:OpenGL::EGL>
+          $<$<AND:$<PLATFORM_ID:Linux,FreeBSD>,$<BOOL:${ENABLE_WAYLAND}>>:Wayland::EGL>)
```
Should've been added, must have slipped through.

PatTheMav commented on libobs/CMakeLists.txt:
```diff
@@ -249,13 +265,15 @@ elseif(OS_MACOS)
   include(cmake/os-macos.cmake)
 elseif(OS_LINUX)
   include(cmake/os-linux.cmake)
-elseif(OS_FREEBSD)
+elseif(OS_FREEBSD OR OS_OPENBSD)
```
Yeah, if both BSDs are considered supported in the same way, then any inclusion of FreeBSD will also require the inclusion of OpenBSD.

RytoEX replied:
Okay.

PatTheMav commented on plugins/obs-outputs/cmake/ftl.cmake:
```diff
@@ -1,77 +1,65 @@
+          ftl-sdk/libftl/ftl-sdk.c
+          ftl-sdk/libftl/ftl_helpers.c
+          ftl-sdk/libftl/ftl_private.h
+          ftl-sdk/libftl/gettimeofday/gettimeofday.c
+          ftl-sdk/libftl/gettimeofday/gettimeofday.h
```
I don't fully understand it either - the header specifies only a single Windows-specific function (`gettimeofday` is provided by `time.h` on POSIX systems) and the other functions are used in the FTL sdk, so the symbols were required by its code on all platforms.

RytoEX replied:
So the equivalent to the old/existing behavior would be:
```cmake
target_sources(
  ftl-sdk
  PRIVATE
  [...]
  ftl-sdk/libftl/gettimeofday/gettimeofday.c
  $<$<PLATFORM_ID:Windows>:ftl-sdk/libftl/gettimeofday/gettimeofday.h>
  [...]
)
```

But I don't know if that is worth keeping that way. I'm leaning towards "we should just change it" if both files are meant to be included as sources on all platforms.

RytoEX commented on cmake/windows/compilerconfig.cmake:
```diff
@@ -53,6 +57,9 @@ endif()
   /W3
   /utf-8
+  /permissive-
+  /Zc:__cplusplus
+  /Zc:preprocessor
```
I think these are MSVC specific.

RytoEX commented on cmake/windows/compilerconfig.cmake:
```diff
@@ -42,9 +47,8 @@ if(CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION VERSION_LESS 10.0.20348)
-set(_obs_msvc_c_options /Brepro /MP /permissive- /Zc:__cplusplus /Zc:preprocessor)
-
-set(_obs_msvc_cpp_options /Brepro /MP /permissive- /Zc:__cplusplus /Zc:preprocessor)
+set(_obs_msvc_c_options /Brepro /MP)
+set(_obs_msvc_cpp_options /Brepro /MP)
```
I believe clang-cl supports `/Brepro`.
https://blog.llvm.org/2019/11/deterministic-builds-with-clang-and-lld.html
https://releases.llvm.org/17.0.1/tools/clang/docs/UsersManual.html#id10

RytoEX commented on cmake/windows/compilerconfig.cmake:
```diff
@@ -53,6 +57,9 @@ endif()
 add_compile_options(
   /W3
   /utf-8
+  /permissive-
```
Clang seems to supports this. Ignore the mention of a revert. They didn't actually revert it, and simply added a release note item.
https://reviews.llvm.org/D103773

RytoEX commented on plugins/text-freetype2/CMakeLists.txt:
```diff
@@ -11,27 +11,52 @@ endif()
+  PRIVATE # cmake-format: sortable
+          $<$<PLATFORM_ID:Darwin>:find-font-cocoa.m>
+          $<$<PLATFORM_ID:Darwin>:find-font-iconv.c>
+          $<$<PLATFORM_ID:Darwin>:find-font.c>
+          $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD>:find-font-unix.c>
+          $<$<PLATFORM_ID:Windows>:find-font-windows.c>
+          $<$<PLATFORM_ID:Windows>:find-font.c>
```
I might have asked before, but why:
```
$<$<PLATFORM_ID:Darwin>:find-font.c>
$<$<PLATFORM_ID:Windows>:find-font.c>
```
Instead of:
```
$<$<PLATFORM_ID:Darwin,Windows>:find-font.c>
```

Not a big deal, just curious.

PatTheMav replied:
Just an oversight when the OS-specific branches were moved into generator expressions.

RytoEX reviewed (changes requested):
Quick review of files I hadn't marked viewed since last review.

PatTheMav commented on UI/frontend-plugins/decklink-captions/CMakeLists.txt:
```diff
@@ -7,31 +7,40 @@ if(NOT ENABLE_DECKLINK)
 endif()
 
-find_qt(COMPONENTS Widgets)
+if(OS_LINUX
+   OR OS_FREEBSD
+   OR OS_OPENBSD)
+  find_package(X11 REQUIRED)
+  mark_as_advanced(X11)
+endif()
```
See above.

RytoEX commented:
> Remove support for Qt5, removed helper function used during transition

I think this was left in for now to ease the transition of submodules to removing their use of it. We can remove the actual CMake function in the next major release. In any case, the PR description may need updating.

PatTheMav commented:
> > Remove support for Qt5, removed helper function used during transition
> 
> I think this was left in for now to ease the transition of submodules to removing their use of it. We can remove the actual CMake function in the next major release. In any case, the PR description may need updating.

Updated the description - Qt5 was indeed removed in the parts that resolve dependencies of the main app but the helper function is still there.

PatTheMav commented:
Rebased and squashed commits into logical units.

RytoEX reviewed (approved):
Local tests seemed fine. CI is green.
